xla/service/cpu/onednn_memory_util.cc

```diff
@@ -90,16 +92,15 @@ void SetDimsStrides(const Shape& shape, std::vector<int64_t>& dims,
       stride *= dims.at(i);
     }
   }
+  return std::make_pair(dims, strides);
 }
 
 StackAlloca GetAllocaAndEmitMemrefInfo(llvm::IRBuilderBase& builder,
                                        const llvm_ir::IrArray& ir_array) {
   const Shape& shape = ir_array.GetShape();
   // oneDNN handles scalar as a vector of size 1.
   int64_t rank = shape.rank() == 0 ? 1 : shape.rank();
-  std::vector<int64_t> dims;
-  std::vector<int64_t> strides(rank);
-  SetDimsStrides(shape, dims, strides);
+  auto dims_strides = GetDimsStrides(shape);
```

Comment:
One last nit please:
We can unpack the pair to two separate variables to improve readability. See below for example:
https://github.com/openxla/xla/blob/cbcde508791510b6bd0593d3ac198378eb95e478/xla/backends/cpu/runtime/sort_thunk.cc#L161

Reply:
Sure. I made the change.